sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala

```diff
@@ -0,0 +1,392 @@
+    val qlTable = toQlTable(hTable)
+    val qlPartitions = version match {
+      case hive.v12 => client.call[Table, Set[Partition]]("getAllPartitionsForPruner", qlTable)
+      case hive.v13 => client.call[Table, Set[Partition]]("getAllPartitionsOf", qlTable)
+    }
+    qlPartitions.map(_.getTPartition).map { p =>
+      HivePartition(
+        values = Option(p.getValues).map(_.toSeq).getOrElse(Seq.empty),
+        storage = HiveStorageDescriptor(
+          location = p.getSd.getLocation,
+          inputFormat = p.getSd.getInputFormat,
+          outputFormat = p.getSd.getOutputFormat,
+          serde = p.getSd.getSerdeInfo.getSerializationLib))
+    }.toSeq
+  }
+
+  def listTables(dbName: String): Seq[String] = withClassLoader {
+    client.getAllTables
+  }
+
+  /**
+   * Runs the specified SQL query using Hive.
+   */
+  def runSqlHive(sql: String): Seq[String] = {
+    val maxResults = 100000
+    val results = runHive(sql, maxResults)
+    // It is very confusing when you only get back some of the results...
+    if (results.size == maxResults) sys.error("RESULTS POSSIBLY TRUNCATED")
+    results
+  }
+
+  /**
+   * Execute the command using Hive and return the results as a sequence. Each element
+   * in the sequence is one row.
+   */
+  protected def runHive(cmd: String, maxRows: Int = 1000): Seq[String] = withClassLoader {
+    logDebug(s"Running hiveql '$cmd'")
+    if (cmd.toLowerCase.startsWith("set")) { logDebug(s"Changing config: $cmd") }
+    try {
+      val cmd_trimmed: String = cmd.trim()
+      val tokens: Array[String] = cmd_trimmed.split("\\s+")
+      val cmd_1: String = cmd_trimmed.substring(tokens(0).length()).trim()
+      val proc: CommandProcessor = version match {
+        case hive.v12 =>
+          classOf[CommandProcessorFactory]
+            .callStatic[String, HiveConf, CommandProcessor]("get", cmd_1, conf)
+        case hive.v13 =>
+          classOf[CommandProcessorFactory]
+            .callStatic[Array[String], HiveConf, CommandProcessor]("get", Array(tokens(0)), conf)
+      }
+
+      proc match {
+        case driver: Driver =>
+          val response: CommandProcessorResponse = driver.run(cmd)
+          // Throw an exception if there is an error in query processing.
+          if (response.getResponseCode != 0) {
+            driver.close()
+            throw new QueryExecutionException(response.getErrorMessage)
+          }
+          driver.setMaxRows(maxRows)
+
+          val results = version match {
+            case hive.v12 =>
+              val res = new JArrayList[String]
+              driver.call[JArrayList[String], Boolean]("getResults", res)
+              res.toSeq
+            case hive.v13 =>
+              val res = new JArrayList[Object]
+              driver.call[JArrayList[Object], Boolean]("getResults", res)
+              res.map { r =>
+                r match {
+                  case s: String => s
+                  case a: Array[Object] => a(0).asInstanceOf[String]
```

Comment:
I honestly have no idea.  This code came from the Hive 13 upgrade, but once again is used only for metadata operations or golden answer generation.